salt/modules/win_pkg.py

```diff
@@ -402,63 +482,208 @@ def _get_local_repo_dir(saltenv='base'):
     else:
         winrepo_source_dir = __opts__['winrepo_source_dir']
 
-    dirs = []
-    dirs.append(salt.syspaths.CACHE_DIR)
-    dirs.extend(['minion', 'files'])
-    dirs.append(saltenv)
-    dirs.extend(winrepo_source_dir[7:].strip('/').split('/'))
-    return os.sep.join(dirs)
+    #dest_path = '{0}\\files\\{1}\\win\\repo-ng'\
+    #    .format(__opts__['cachedir'], saltenv)
+
+    dirs = [__opts__['cachedir'], 'files', saltenv]
+    url_parts = _urlparse(winrepo_source_dir)
+    dirs.append(url_parts.netloc)
+    dirs.extend(url_parts.path.strip('/').split('/'))
+    dest_path = os.sep.join(dirs)
+    return (winrepo_source_dir, dest_path)
 
 
-def genrepo(saltenv='base'):
+def _repo_age(saltenv):
     '''
-    Generate winrepo_cachefile based on sls files in the winrepo
+    Returns age in seconds of the generated repo
+    '''
+    (repo_remote, repo_local) = _get_repo_src_dest(saltenv)
+    if not os.path.exists(repo_local):
+        os.makedirs(repo_local)
+    winrepo = os.path.join(repo_local, 'winrepo.p')
+
+    try:
```

Comment:
What is the point of a try/except here, if all we do is raise the exception? :confused: